apis/datadoghq/v2alpha1/datadogagent_default.go

```diff
@@ -0,0 +1,362 @@
+// Note: many default values are set in the Datadog Agent and deliberately not set by the Operator.
+const (
+	defaultClusterName = "datadog-monitoring"
+
+	defaultLogCollectionEnabled          bool   = false
+	defaultLogContainerCollectUsingFiles bool   = true
+	defaultLogContainerLogsPath          string = "/var/lib/docker/containers"
+	defaultLogPodLogsPath                string = "/var/log/pods"
+	defaultLogContainerSymlinksPath      string = "/var/log/containers"
+	defaultLogTempStoragePath            string = "/var/lib/datadog-agent/logs"
+
+	defaultLiveProcessCollectionEnabled   bool = false
+	defaultLiveContainerCollectionEnabled bool = true
+
+	defaultOOMKillEnabled        bool = true
+	defaultTCPQueueLengthEnabled bool = false
+
+	defaultAPMEnabled         bool   = false
+	defaultAPMHostPortEnabled bool   = false
+	defaultAPMHostPort        int32  = 8126
+	defaultAPMSocketEnabled   bool   = true
+	defaultAPMSocketPath      string = "/var/run/datadog/apm.sock"
+
+	defaultCSPMEnabled              bool = false
+	defaultCWSEnabled               bool = false
+	defaultCWSSyscallMonitorEnabled bool = false
+
+	defaultNPMEnabled         bool = false
+	defaultNPMEnableConntrack bool = true
+	defaultNPMCollectDNSStats bool = true
+
+	defaultUSMEnabled bool = false
+
+	defaultDogStatsDOriginDetectionEnabled bool   = false
+	defaultDogStatsDHostPortEnabled        bool   = false
+	defaultDogStatsDPort                   int32  = 8125
+	defaultDogStatsDUseSocketVolume        bool   = false
+	defaultDogStatsDSocketPath             string = "/var/run/datadog/dsd.socket"
+
+	defaultCollectKubernetesEvents bool = true
+
+	defaultAdmissionControllerEnabled          bool = false
+	defaultAdmissionControllerMutateUnlabelled bool = false
+
+	defaultOrchestratorExplorerEnabled bool   = true
+	DefaultOrchestratorExplorerConf    string = "orchestrator-explorer-config"
+
+	defaultExternalMetricsServerEnabled bool = false
+	defaultDatadogMetricsEnabled        bool = true
+	// Cluster Agent versions < 1.20 should use 443
+	defaultMetricsProviderPort int32 = 8443
+
+	defaultKubeStateMetricsCoreEnabled bool   = false
+	defaultKubeStateMetricsCoreConf    string = "kube-state-metrics-core-config"
+
+	defaultClusterChecksEnabled    bool = true
+	defaultUseClusterChecksRunners bool = true
+
+	defaultPrometheusScrapeEnabled                bool = false
+	defaultPrometheusScrapeEnableServiceEndpoints bool = false
+
+	defaultDatadogMonitorEnabled bool = false
+)
+
+// DefaultDatadogAgent defaults the DatadogAgentSpec GlobalConfig and Features.
+func DefaultDatadogAgent(dda *DatadogAgent) {
+	defaultGlobalConfig(&dda.Spec)
+
+	defaultFeaturesConfig(&dda.Spec)
+}
+
+// defaultGlobalConfig sets default values in DatadogAgentSpec.Global.
+func defaultGlobalConfig(ddaSpec *DatadogAgentSpec) {
+	if ddaSpec.Global.ClusterName == nil {
+		defaultClusterNameCopy := defaultClusterName
```

Comment:
I imagine that we have a lot of those, could just a `func()` that returns the pointer